src/pages/data-feeds/starknet/index.md

```diff
@@ -0,0 +1,147 @@
+section: ethereum
+date: Last Modified
+title: "StarkNet"
+metadata:
+  description: "Using Chainlink Data Feeds on StarkNet"
+---
+
+StarkNet is a permissionless decentralized ZK-Rollup operating as an L2 network over Ethereum. Unlike other Ethereum L2 networks, StarkNet uses [Cairo](https://www.cairo-lang.org/docs/index.html) as its smart contract language. Chainlink Data Feeds are available on the StarkNet testnet as Cairo smart contracts.
+
+For a complete list of Chainlink Price Feeds available on StarkNet testnet, see the [Price Feed Contract Addresses](https://docs.chain.link/data-feeds/price-feeds/addresses?network=starknet) page.
+
+## Using Data Feeds on StarkNet
+
+You can read Chainlink Data Feeds on StarkNet using an [on-chain contract](#running-the-on-chain-example) that you compile and deploy or [read the data feed off-chain](#running-the-off-chain-example). These examples use [StarkNet.js](https://www.starknetjs.com/), [HardHat](https://hardhat.org/), and the [StarkNet Hardhat Plugin](https://shard-labs.github.io/starknet-hardhat-plugin/) to compile and deploy contracts. Reading data feeds off chain can be done using only the [StarkNet CLI](https://docs.starknet.io/documentation/tools/CLI/commands/) and the ABI for the data feed proxy contract, but these examples also show off-chain reads using [StarkNet.js](https://www.starknetjs.com/) without a StarkNet account.
+
+### Requirements
+
+Set up your environment to run the examples.
+
+1. [Setup your local StarkNet environment](https://starknet.io/docs/quickstart.html). Note that a Python version in the `>=3.6 <=3.9` range is required for compiling and deploying contracts on-chain. The [`cairo-lang` Python package](https://pypi.org/project/cairo-lang/) is not compatible with newer versions of Python as of the [`cairo-lang` 0.10.3](https://pypi.org/project/cairo-lang/0.10.3/) package. Check [starknet.io](https://starknet.io/docs/quickstart.html) for the latest requirements.
+1. [Set up a StarkNet account](https://starknet.io/docs/hello_starknet/account_setup.html) on StarkNet's `alpha-goerli` network and fund it with [testnet ETH](https://faucet.goerli.starknet.io/). These examples expect the OpenZeppelin wallet, which stores your addresses and private keys at `~/.starknet_accounts/starknet_open_zeppelin_accounts.json` by default.
+1. [Install NodeJS](https://nodejs.org/en/download/) in the version in the `>=14 <=18` version range.
+1. [Install Yarn](https://classic.yarnpkg.com/lang/en/docs/install/).
+1. Clone the [smartcontractkit/chainlink-starknet](https://github.com/smartcontractkit/chainlink-starknet) repository, which includes the example contracts for this guide: `git clone https://github.com/smartcontractkit/chainlink-starknet.git`
+1. In your clone of the [chainlink-starknet](https://github.com/smartcontractkit/chainlink-starknet) repository, change directories to the proxy consumer example: `cd ./chainlink-starknet/examples/contracts/proxy-consumer/`
```

Comment:
Nit: Put these longer commands in a code block indented within the steps rather than inline. (Can't guarantee my indentations would be correct in a `suggestion` block so I omitted it.)

Reply:
Done